includes/import-products.php

```diff
@@ -18,11 +18,26 @@
  * @throws Exception
  */
 function dw_import_products() {
+	global $dw_options;
+
 	$client  = RestApiClientFactory::getClient();
 	$courses = $client->getCourseInstances(new DateTime(), null) ?? [];
-
 	$createdProducts = [];
 
+	if( $dw_options['ccvCode'] && $dw_options['ccvCode'] !== "") {
+		$codesToFilter = explode(';', $dw_options['ccvCode']);
+
+		$courses = array_filter($courses, function($course) use ($codesToFilter) {
+			foreach($codesToFilter as $code) {
```

Comment:
If you array_filter the `$codesToFilter` array once and remove the empty string, you then only have to do:

```
$codesToFilter = array_filter(
	explode(';', $dw_options['ccvCode']),
	function ($code) {
		return $code !== '';
	}
);

$courses = array_filter($courses, function($course) use ($codesToFilter) {
	return in_array($course['ccvCode'], $codesToFilter);
});
```